openghg/retrieve/icos/_retrieve.py

```diff
@@ -0,0 +1,223 @@
+
+        # How to best handle this? Static method seems ?
+        obs = ObsSurface.load()
+        obs.store_data(data=standardised_data)
+        obs.save()
+
+        # Create the expected ObsData type
+        obs_data = []
+        for data in standardised_data.values():
+            measurement_data = data["data"]
+            metadata = data["metadata"]
+            obs_data.append(ObsData(data=measurement_data, metadata=metadata))
+
+        if len(obs_data) == 1:
+            return obs_data[0]
+        else:
+            return obs_data
+
+
+def _retrieve_remote(
+    site: str,
+    species: Optional[Union[str, List]] = None,
+    sampling_height: Optional[str] = None,
+) -> Dict:
+    """Retrieve ICOS data from the ICOS Carbon Portal and standardise it into
+    a format expected by OpenGHG. A dictionary of metadata and Datasets
+
+    Args:
+        site: ICOS site code, for site codes see
+        https://www.icos-cp.eu/observations/atmosphere/stations
+        sampling_height: Sampling height in metres
+    Returns:
+        dict: Dictionary of processed data and metadata
+    """
+    from icoscp.station import station  # type: ignore
+    from icoscp.cpb.dobj import Dobj  # type: ignore
+    from openghg.standardise.meta import assign_attributes
+    from openghg.util import load_json
+    import re
+
+    if species is None:
+        species = ["CO", "CO2", "N2O"]
+
+    if not isinstance(species, list):
+        species = [species]
+
+    # We should first check if it's stored in the object store
+    # Will need to make sure ObsSurface can accept the datasets we
+    # create from the ICOS data
+    stat = station.get(stationId=site.upper())
+    # See https://icos-carbon-portal.github.io/pylib/modules/#stationdatalevelnone
+    # - Data level 2: The final quality checked ICOS RI data set, published by the CFs,
+    #  to be distributed through the Carbon Portal.
+    data_level = 2
+    data_pids = stat.data(level=data_level)
+
+    # We want to get the PIDs of the data for each species here
+    species_upper = [s.upper() for s in species]
+    # For this see https://stackoverflow.com/a/55335207
+    search_str = r"\b(?:{})\b".format("|".join(map(re.escape, species_upper)))
+    # Now filter the dataframe so we can extraxt the PIDS
+    filtered_sources = data_pids[data_pids["specLabel"].str.contains(search_str)]
+
+    if sampling_height is not None:
+        sampling_height = str(float(sampling_height.rstrip("m")))
+        filtered_sources = filtered_sources[
+            [sampling_height in x for x in filtered_sources["samplingheight"]]
+        ]
+
+    # Now extract the PIDs along with some data about them
+    dobj_urls = filtered_sources["dobj"].tolist()
+
+    site_metadata = load_json("icos_atmos_site_metadata.json")
+
+    standardised_data: Dict[str, Dict] = {}
+
+    for dobj_url in dobj_urls:
+        dobj = Dobj(dobj_url)
+
+        print(dobj)
+
+        continue
```

Comment:
Was this `continue` left in by mistake? This stops the loop from actually retrieving the data and the moment and just an empty list is returned.